### ISSUE TITLE
Re-Introspection: `referentialIntegrity = prisma` not respected when using `@@map()`

### ISSUE THREAD

**Issue by SamuelQuinones** [prisma/prisma#11022]
### Bug description

I have the following schema for a database (see below)

and when i run `prisma db pull` all of my relations are removed.

however when i remove the use of the `@@map()` function call, the relations are maintained.  I am unsure if this is by design or not?

I created the database before i knew what prisma was and we have a few stored procedures so updating the table names would not be without a headache

### How to reproduce

1. create a schema 
2. enable the `referentialIntegrity` feature and set it to `prisma`
3. create a schema using maps
4.  run `prisma generate`
5. run `prisma db push`
6. run `prisma db pull` and see the relations are removed

### Expected behavior

relations are maintained even when using the `@@map()` function

### Prisma information

```prisma
generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["referentialIntegrity"]
}

datasource db {
  provider             = "mysql"
  url                  = env("DATABASE_URL")
  referentialIntegrity = "prisma"
}

model MemberRole {
  id      Int      @id @default(autoincrement())
  name    String   @unique(map: "role_name_unique") @db.VarChar(255)
  members Member[]

  @@map("member_roles_friendly")
}

model Member {
  id                 Int        @id @default(autoincrement())
  active             Boolean    @default(true)
  name               String     @db.VarChar(50)
  photo              String?    @db.Text
  github             String?    @db.Text
  youtube            String?    @db.Text
  twitter            String?    @db.Text
  website            String?    @db.Text
  role_id            Int
  role               MemberRole @relation(fields: [role_id], references: [id])
  coreteam           Boolean    @default(false)
  joined_timestamp   DateTime?  @default(now()) @db.Timestamp(0)
  inactive_timestamp DateTime?  @db.Timestamp(0)
  modified_timestamp DateTime?  @default(now()) @db.Timestamp(0)

  @@map("members")
}

model Repository {
  id                 Int       @id @default(autoincrement())
  repo_name          String    @unique(map: "repo_name_unique") @db.VarChar(255)
  repo_link          String?   @db.Text
  modified_timestamp DateTime  @default(now()) @db.Timestamp(0)
  web_api_support    Boolean   @default(false)
  releases           Release[]

  @@map("release_repo_friendly")
}

model ReleaseResource {
  id            Int     @id @default(autoincrement())
  release_id    Int
  release       Release @relation(fields: [release_id], references: [id])
  name          String  @db.VarChar(255)
  download_link String  @db.Text

  @@map("release_resources")
}

model Release {
  id                 Int               @id @default(autoincrement())
  version            String            @db.VarChar(255)
  is_pre_release     Boolean           @default(false)
  repo_id            Int
  repo               Repository        @relation(fields: [repo_id], references: [id])
  release_page       String            @db.Text
  release_timestamp  DateTime          @default(now()) @db.Timestamp(0)
  modified_timestamp DateTime          @default(now()) @db.Timestamp(0)
  resources          ReleaseResource[]

  @@map("releases")
}
```

### Environment & setup

- OS: Mac OS, Windows
- Database: MYSQL
- Node.js version: Node v 16.13.0

### Prisma Version

```
prisma                  : 3.7.0
@prisma/client          : 3.7.0
Current platform        : darwin-arm64
Query Engine (Node-API) : libquery-engine 8746e055198f517658c08a0c426c7eec87f5a85f (at node_modules/@prisma/engines/libquery_engine-darwin-arm64.dylib.node)
Migration Engine        : migration-engine-cli 8746e055198f517658c08a0c426c7eec87f5a85f (at node_modules/@prisma/engines/migration-engine-darwin-arm64)
Introspection Engine    : introspection-core 8746e055198f517658c08a0c426c7eec87f5a85f (at node_modules/@prisma/engines/introspection-engine-darwin-arm64)
Format Binary           : prisma-fmt 8746e055198f517658c08a0c426c7eec87f5a85f (at node_modules/@prisma/engines/prisma-fmt-darwin-arm64)
Default Engines Hash    : 8746e055198f517658c08a0c426c7eec87f5a85f
Studio                  : 0.445.0
Preview Features        : referentialIntegrity
```

**Comment by janpio** [comment]
I can confirm this with this more minimal schema:
```prisma
generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["referentialIntegrity"]
}

datasource db {
  provider             = "mysql"
  url                  = env("DATABASE_URL")
  referentialIntegrity = "prisma"
}

model Foo {
  id     Int @id
  bar    Bar @relation(fields: [bar_id], references: [id])
  bar_id Int @unique

  @@map("foo_db")
}

model Bar {
  id  Int  @id
  foo Foo?
}
```

After Re-Introspection, we only get this output and schema:

```
Datasource "db": MySQL database "11022" at "g0a52uhg4117.us-east-3.psdb.cloud:3306"

Introspecting based on datasource defined in prisma\schema.prisma …

✔ Introspected 2 models and wrote them into prisma\schema.prisma in 3.33s

*** WARNING ***

These models were enriched with `@@map` information taken from the previous Prisma schema.
- Model "Foo"

Run prisma generate to generate Prisma Client.
```

```prisma
generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["referentialIntegrity"]
}

datasource db {
  provider             = "mysql"
  url                  = env("DATABASE_URL")
  referentialIntegrity = "prisma"
}

model Foo {
  id     Int @id
  bar_id Int @unique

  @@map("foo_db")
}

model Bar {
  id Int @id
}

```

The relation and backrelation are gone.

**Comment by Jolg42** [comment]
Note: fix will be available in next 4.6.0 release planned on Tuesday (November 8th).
